packages/dapp-toolkit/src/modules/wallet-request/request-items/request-item.module.ts

```diff
@@ -38,21 +47,36 @@ export const RequestItemModule = (input: RequestItemModuleInput) => {
     isOneTimeRequest,
   })
 
-  const add = (value: {
-    type: RequestItem['type']
-    walletInteraction: WalletInteraction
-    isOneTimeRequest: boolean
-  }) => {
+  const add = (
+    value: {
+      type: RequestItem['type']
+      walletInteraction: WalletInteraction
+      isOneTimeRequest: boolean
+    },
+    onSignal?: (signalValue: string) => void,
```

Comment:
are signals expected to be used for other things than txId callback? Why pick a generic name for it?